### PR TITLE
#163424172 create office 

### PR DESCRIPTION
**Pull request by haywhyze**
#### What does this PR do?
Implement create a political office 
#### Description of Task to be completed?
- write tests for the functionalities of the create office endpoint
- add a controller class to handle `/offices` endpoint
- modify isEmpty middleware to accomodate `/offices`
- add new middlewares to handle user input.
- create a data model for the office entity.
- add the new controller and neccessary middlewares to the router
#### How should this be manually tested?
- clone this repository by executing `git clone https://github.com/haywhyze/politico.git`
- cd to the cloned local repository `cd politico`
- run `ft-create-office-163424172`
- install all dependencies by running `npm install`
- start the server with `npm start`
- when the server is up, you can test using Postman or any other related service.

#### What are the relevant pivotal tracker stories?
[#163424172](https://www.pivotaltracker.com/story/show/163424172)

### CODE REVIEW

File: server/middlewares/isEmpty.js [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
-const populateError = req => {
+const populateError = (req, ...fields) => {
   const error = [];
-  if (!req.body.name) {
-    error.push('name');
-  }
-  if (!req.body.hqAddress) {
-    error.push('hqAddress');
-  }
+  // eslint-disable-next-line array-callback-return
```
Comment (akinwale-habibullah) [inline server/middlewares/isEmpty.js]: Please use rules in .eslintrc file instead of specifying your rules in individual files.

Comment (haywhyze) [reply server/middlewares/isEmpty.js]: done!

File: server/middlewares/validateType.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const isValidType = (req, res, next) => {
+  const acceptedType = ['federal', 'legislature', 'state', 'local'];
+
+  if (!acceptedType.includes(req.body.type)) {
+    return res.status(400).send({
+      status: 400,
+      error: 'Type not of accepted value',
```
Comment (akinwale-habibullah) [inline server/middlewares/validateType.js]: Please specify the allowed types, so you don't leave the user confused.

**Review by akinwale-habibullah — APPROVED** [review]

File: server/middlewares/validateType.js (hunk 2 above)
Comment (haywhyze) [reply server/middlewares/validateType.js]: okay

**Comment by akinwale-habibullah** [comment]
Please implement my feedback, commit and then go ahead to merge.

**Pull request merged**